Bookish/Controllers/HomeController.cs

```diff
@@ -78,17 +78,50 @@ public ActionResult AddBook(BookModel newBook)
         return View(newBook);
     }
 
-    [HttpPost]
+    [HttpGet]
     public IActionResult ViewCopies(BookModel book)
     {
-        var copies = BookCopyService.GetCopies(book);
-        var result = new BookCopyResult()
+        if (book == null || book.Id == null)
         {
-            book = book,
-            copies = copies
-        };
+            return RedirectToAction(nameof(Catalogue));
+        }
+        var result = BookCopyService.GetCopies(book);

```

Comment:
I like that this logic lives in a service! However the service itself should probably not live in the Models folder, but in a separate Services folder. I left some similar comments on Claire's PR about some of the classes which contained useful methods. I suggested renaming the classes to something like `EditBooksHelper` and putting them in a Helpers folder, but you two should probably chat and come up with a common solution.